### PR TITLE
Export `CMAKE_PREFIX_PATH` in helper installation scripts

### PR DESCRIPTION
**Pull request by francesco-ballarin**
Mimicks the already existing export of `PKG_CONFIG_PATH` and helps downstream development in the nightly docker image.

This change is not required when building the docker nightly image itself, but I honestly do not understand why:
* before this change, https://github.com/FEniCS/dolfinx/blob/main/docker/Dockerfile.end-user#L102 did not set either `CMAKE_PREFIX_PATH` or `DOLFINX_DIR`, yet (as manually verified with a few runs) the complex installation gets correctly detected.
* https://github.com/FEniCS/dolfinx/blob/main/docker/Dockerfile.end-user#L87 does not even source `/usr/local/dolfinx-real/lib/dolfinx/dolfinx.conf`, yet the real installation gets correctly found, even if it is in a non standard path.

I also have verified that without `CMAKE_PREFIX_PATH` (and without `DOLFINX_DIR`) one could have had mixed configurations when using the nightly image, in which after sourcing `/usr/local/dolfinx-complex/lib/dolfinx/dolfinx.conf` the python part is imported from `/usr/local/dolfinx-complex`, while the C++ one from `/usr/local/dolfinx-real`.

### CODE REVIEW

**Review by jhale — APPROVED** [review]

**Comment by jhale** [comment]
It isn't strictly necessary to be up to date with main to merge. The merge queue will handle merging main into your branch and running the test suite.

**Comment by francesco-ballarin** [comment]
I'll wait to see if @garth-wells has any comment, and after that I'll merge to main.

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -29,15 +29,15 @@ execute_process(
   ERROR_VARIABLE BASIX_ERROR_OUT OUTPUT_STRIP_TRAILING_WHITESPACE
 )
 find_package(Basix REQUIRED CONFIG HINTS ${BASIX_PY_DIR})
+message(STATUS "Found Basix at ${BASIX_DIR}")
```
Comment (garth-wells) [inline python/CMakeLists.txt:32]: This needs an if/else, e.g.
```
if (Basix_FOUND)
  message(STATUS "Found Basix at ${BASIX_DIR}")
endif()
```

File: python/CMakeLists.txt [hunk 2]
```diff
@@ -29,15 +29,15 @@ execute_process(
...
 find_package(DOLFINX REQUIRED CONFIG)
+message(STATUS "Found DOLFINx at ${DOLFINX_DIR}")
```
Comment (garth-wells) [inline python/CMakeLists.txt:34]: See above comment on if/else.

**Review by garth-wells — CHANGES_REQUESTED** [review]

File: python/CMakeLists.txt (hunk 1 above)
Comment (francesco-ballarin) [reply python/CMakeLists.txt:32]: Why is that? My understanding is that `find_package(... REQUIRED ...)` will stop with an error if the package is not found, and therefore the line with message will only get printed if the package was actually found.

Comment (garth-wells) [reply python/CMakeLists.txt:32]: I don't think CMake necessarily stops immediately. It's also a readability issue.

Comment (francesco-ballarin) [reply python/CMakeLists.txt:32]: OK, will do!

**Review by garth-wells — APPROVED** [review]

**Pull request merged**